plugins/ufm_log_analyzer_plugin/src/loganalyze/log_analyzers/console_log_analyzer.py

```diff
@@ -22,11 +24,54 @@ class ConsoleLogAnalyzer(BaseAnalyzer):
     def __init__(
         self, logs_csvs: List[str], hours: int, dest_image_path, sort_timestamp=True
     ):
+        self.ufm_versions = self._extract_ufm_version(logs_csvs)
         self.fix_lines_with_no_timestamp(logs_csvs)
         super().__init__(logs_csvs, hours, dest_image_path, sort_timestamp)
         self._log_data_sorted.dropna(subset=["data"], inplace=True)
         self._funcs_for_analysis = {self.print_exceptions_per_time_count}
 
+    @staticmethod
+    def _extract_ufm_version(logs_csvs):
+        """
+        This function gets all the ufm version from the console log.
+        It saves then in a set and returns all the unique UFM version
+        that were found in the log
+        """
+        ufm_versions = set()  # List to store ufm_version rows
+
+        for csv_file in logs_csvs:
+            temp_file = csv_file + ".temp"
+
+            # Open the input CSV file for reading
+            with open(csv_file,
```

Comment:
Is logs_csvs guaranteed to contain valid file paths? It may be useful to check if the file exists before processing it, or handle cases where the file might be missing

Reply:
Good question.

Since I'm the one who creates them, and fully manages them. I know they exists

Reply:
Ok